Feature/docker container registry

vdevic01 opened the pull request:
@miloss01 @SelenaMilutin I will need your help with this one. Part of the model with docker image and tags was slightly changed. I updated the tests and deleted tags attribute from DockerImageDTO but I believe there is a miss understanding about the way model is used on the frontend. DockerImageDTO currently contains some attributes that are not connected to image but rather repository (Badge, Description, Star Count). On the frontend we should show available tags (+image data of image connected to that tag (last push, digest...)) in repository section and repositories in explore section.

Closes #34

miloss01 commented:
> @miloss01 @SelenaMilutin I will need your help with this one. Part of the model with docker image and tags was slightly changed. I updated the tests and deleted tags attribute from DockerImageDTO but I believe there is a miss understanding about the way model is used on the frontend. DockerImageDTO currently contains some attributes that are not connected to image but rather repository (Badge, Description, Star Count). On the frontend we should show available tags (+image data of image connected to that tag (last push, digest...)) in repository section and repositories in explore section.
> 
> Closing #34

About DockerImageDTO, as I remember, when I made that DTO, I put repo data you mentioned in it because there was no dto for repo and I didn't want to make one since I didn't really need it and didn't want to double it if someone else made it. Also, I used DockerImage.ID as digest value (it probably won't be hard to change that).

In Explore section, listed items are docker images containing data about their tags (it was array of strings as I remember) amongst other data. As I see it, in that moment it was basically what you are saying now. What I'm saying, I guess, is that listed items are docker images with tags as data and not tags with docker images as data. Honestly, it's fine with me to leave it like that since I don't really see much of a difference between these two approaches.

At first look, I wouldn't change much. It's needed to swap ID with Digest property on frontend. Also, I wouldn't change DockerImageDTO much since it would be a little bit of a hassle to sort it out now. What I see that it's needed to change is to add Digest property to DockerImageDTO and to not just pass Tags attribute from model to dto as I was doing until now but to get tag names connected to that image and convert it to string array. That way dto is not changed much, also frontend would basically look the same but with ID and Digest properties swapped.

Does that sound ok?

vdevic01 commented:
> > @miloss01 @SelenaMilutin I will need your help with this one. Part of the model with docker image and tags was slightly changed. I updated the tests and deleted tags attribute from DockerImageDTO but I believe there is a miss understanding about the way model is used on the frontend. DockerImageDTO currently contains some attributes that are not connected to image but rather repository (Badge, Description, Star Count). On the frontend we should show available tags (+image data of image connected to that tag (last push, digest...)) in repository section and repositories in explore section.
> > Closing #34
> 
> About DockerImageDTO, as I remember, when I made that DTO, I put repo data you mentioned in it because there was no dto for repo and I didn't want to make one since I didn't really need it and didn't want to double it if someone else made it. Also, I used DockerImage.ID as digest value (it probably won't be hard to change that).
> 
> In Explore section, listed items are docker images containing data about their tags (it was array of strings as I remember) amongst other data. As I see it, in that moment it was basically what you are saying now. What I'm saying, I guess, is that listed items are docker images with tags as data and not tags with docker images as data. Honestly, it's fine with me to leave it like that since I don't really see much of a difference between these two approaches. I think this applies to repo section as well.
> 
> At first look, I wouldn't change much. It's needed to swap ID with Digest property on frontend. Also, I wouldn't change DockerImageDTO much since it would be a little bit of a hassle to sort it out now. What I see that it's needed to change is to add Digest property to DockerImageDTO and to not just pass Tags attribute from model to dto as I was doing until now but to get tag names connected to that image and convert it to string array. That way dto is not changed much, also frontend would basically look the same but with ID and Digest properties swapped.
> 
> Does that sound ok?
> 
> PS Since we have 3 PRs opened, I suggest to close this one last since it may make major(ish) changes.

Regarding explore section I think that page should display repositories and not images since one repository can have hundreds of similar images. Regarding DTO I'm fine with either approach (docker image with list of tags or tag with docker image id) as long as all necessary data is passed to the front.

miloss01 commented:
> > > @miloss01 @SelenaMilutin I will need your help with this one. Part of the model with docker image and tags was slightly changed. I updated the tests and deleted tags attribute from DockerImageDTO but I believe there is a miss understanding about the way model is used on the frontend. DockerImageDTO currently contains some attributes that are not connected to image but rather repository (Badge, Description, Star Count). On the frontend we should show available tags (+image data of image connected to that tag (last push, digest...)) in repository section and repositories in explore section.
> > > Closing #34
> > 
> > 
> > About DockerImageDTO, as I remember, when I made that DTO, I put repo data you mentioned in it because there was no dto for repo and I didn't want to make one since I didn't really need it and didn't want to double it if someone else made it. Also, I used DockerImage.ID as digest value (it probably won't be hard to change that).
> > In Explore section, listed items are docker images containing data about their tags (it was array of strings as I remember) amongst other data. As I see it, in that moment it was basically what you are saying now. What I'm saying, I guess, is that listed items are docker images with tags as data and not tags with docker images as data. Honestly, it's fine with me to leave it like that since I don't really see much of a difference between these two approaches. I think this applies to repo section as well.
> > At first look, I wouldn't change much. It's needed to swap ID with Digest property on frontend. Also, I wouldn't change DockerImageDTO much since it would be a little bit of a hassle to sort it out now. What I see that it's needed to change is to add Digest property to DockerImageDTO and to not just pass Tags attribute from model to dto as I was doing until now but to get tag names connected to that image and convert it to string array. That way dto is not changed much, also frontend would basically look the same but with ID and Digest properties swapped.
> > Does that sound ok?
> > PS Since we have 3 PRs opened, I suggest to close this one last since it may make major(ish) changes.
> 
> Regarding explore section I think that page should display repositories and not images since one repository can have hundreds of similar images. Regarding DTO I'm fine with either approach (docker image with list of tags or tag with docker image id) as long as all necessary data is passed to the front.

Yeah, I guess it makes sense. I will change it.

It seams to me that it would be best to first merge my PR with tests, then analytics PR and then this PR. After that I can make branch to adapt my functionalities to new model and fix explore section and connected features and tests (if needed).

If you agree, you can review my PR so it can be merged anytime if it's ok.
